hazelcast/src/main/java/com/hazelcast/config/AbstractCacheConfig.java

```diff
@@ -46,25 +48,20 @@
 
     private static final String DEFAULT_KEY_VALUE_TYPE = "java.lang.Object";
 
-    /**
-     * The {@link CacheEntryListenerConfiguration}s for the {@link javax.cache.configuration.Configuration}.
-     */
-    protected Set<CacheEntryListenerConfiguration<K, V>> listenerConfigurations;
-
     /**
      * The {@link javax.cache.configuration.Factory} for the {@link javax.cache.integration.CacheLoader}.
      */
-    protected Factory<CacheLoader<K, V>> cacheLoaderFactory;
+    protected DeferredValue<Factory<CacheLoader<K, V>>> lazyCacheLoaderFactory;
 
     /**
      * The {@link Factory} for the {@link javax.cache.integration.CacheWriter}.
      */
-    protected Factory<CacheWriter<? super K, ? super V>> cacheWriterFactory;
+    protected DeferredValue<Factory<CacheWriter<? super K, ? super V>>> cacheWriterFactory;
```

Comment:
Minor: You can also rename the other `DeferredValue`s to lazy or revert the lazy one for consistency